internal/store/job.go

```diff
@@ -163,16 +164,49 @@ var (
 		),
 		*generator.NewFamilyGenerator(
 			"kube_job_status_failed",
-			"The number of pods which reached Phase Failed.",
+			"The number of pods which reached Phase Failed & their failure reason.",
 			metric.Gauge,
 			"",
 			wrapJobFunc(func(j *v1batch.Job) *metric.Family {
-				return &metric.Family{
-					Metrics: []*metric.Metric{
-						{
-							Value: float64(j.Status.Failed),
+				var ms []*metric.Metric
+
+				if float64(j.Status.Failed) == 0 {
+					return &metric.Family{
+						Metrics: []*metric.Metric{
+							{
+								Value: float64(j.Status.Failed),
+							},
 						},
-					},
+					}
+				}
+
+				for _, condition := range j.Status.Conditions {
+
+					if condition.Type == v1batch.JobFailed {
+						reasonKnown := false
+						for _, reason := range jobFailureReasons {
+							reasonKnown = reasonKnown || failureReason(&condition, reason)
+
```

Comment:
You are missing an if check here against `reasonKnown` no, similar to the unknown reasons?

Reply:
Hey @lilic 
I don't think we need an `if` check here because we need to print for all the known reasons with `0` value. Suppose, we get a known reason `BackoffLimitExceeded` then we will be printing
```
kube_job_status_failed{condition="Failed",job_name="FailedJob1",namespace="ns1",reason="BackoffLimitExceeded"} 1 
kube_job_status_failed{condition="Failed",job_name="FailedJob1",namespace="ns1",reason="DeadLineExceeded"} 0
kube_job_status_failed{condition="Failed",job_name="FailedJob1",namespace="ns1",reason="Evicted"} 0
```

and if we get an unknown reason then we will be printing 

```
kube_job_status_failed{condition="Failed",job_name="FailedJob1",namespace="ns1",reason=""} 1
kube_job_status_failed{condition="Failed",job_name="FailedJob1",namespace="ns1",reason="BackoffLimitExceeded"} 0 
kube_job_status_failed{condition="Failed",job_name="FailedJob1",namespace="ns1",reason="DeadLineExceeded"} 0
kube_job_status_failed{condition="Failed",job_name="FailedJob1",namespace="ns1",reason="Evicted"} 0
```

Let me know if we need to make any changes.

Reply:
@lilic - Should I resolve this conversation?